backend/app/models/user.rb

```diff
@@ -4,6 +4,175 @@
 class User < ActiveRecord::Base
   has_secure_password
 
+  validates :name,             presence: true
+  validates :login,            presence: true, uniqueness: { case_sensitive: false }
+  validates :password,         presence: true
+  validate  :custom_validator
+
+  # Return all users.
+  #
+  # @return all users
+  def self.find_all
+    User.select("id, name, login, created_at, permission_ids").order(:name)
+  end
+
+  # Return all users beginning with fitst letter l ('*' as non-alphanumeric wildcard).
+  #
+  # @return all users beginning with fitst letter l ('*' as non-alphanumeric wildcard)
+  def self.find_letter(l)
```

Comment:
this name suggests you are searching for the letter rather than users whose name starts with the letter

Reply:
Done.